src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -936,3 +966,98 @@ def get_entry_to_persist(self, sp_id, tenant):
             entry[_SERVICE_PRINCIPAL_CERT_THUMBPRINT] = self.thumbprint
 
         return entry
+
+
+class ClientRedirectServer(BaseHTTPServer.HTTPServer):  # pylint: disable=too-few-public-methods
+    query_params = {}
+
+
+class ClientRedirectHandler(BaseHTTPServer.BaseHTTPRequestHandler):
+    # pylint: disable=line-too-long
+
+    # backup greeting pages if official ones from "docs.microsoft.com" are unavailable
+    ok_html = ('<!DOCTYPE html>'
```

Comment:
I'd recommend extracting this out into a separate HTML file for easier maintenance.